### DISCUSSION TITLE
Retrieving socket after binding to port 0

### DISCUSSION THREAD

**Discussion by michalwarda** [uNetworking/uWebSockets#1577] [Q&A]
I'm trying to create a server with port 0 as the parameter to get a dynamically assigned port. The problem though is that I'm not sure how to retrieve the port then. Any suggestions?

And if it's important I'm using `capi`.

**Comment by uNetworkingAB** [comment]
https://github.com/uNetworking/uSockets/blob/5d1d640968fb9361575b8454605c1fdc119588c3/src/libusockets.h#L315

Ssl is 0 or 1

**Reply by 0xmemorygrinder** [reply]
Hello,

What about those who does not use the capi ? Is there any method to get the same information in the c++ wrapper ? 

Thanks in advance for your answer

**Reply by uNetworkingAB** [reply]
It's actually not a C++ wrapper, it's a C wrapper. The native library is C++

**Reply by uNetworkingAB** [reply]
Ah okay I see. This particular function is C. There is no C++ wrapper for uSockets.

**Reply by 0xmemorygrinder** [reply]
Alright I can use the C api but how do I get the `struct us_socket_t *` when using the c++ examples ?

**Reply by uNetworkingAB** [reply]
Okay I see. You can pass the listen socket us_listen_socket_t as us_socket_t. You get the listen socket from app.listen

**Reply by 0xmemorygrinder** [reply]
Excellent thanks for your answer !

**Comment by ddevienne** [comment]
In case that helps anyone:
```C++
    inline constexpr bool wSSL = false;
    app.listen(
        port, [&ctx, port](us_listen_socket_t* socket) -> void {
            if (socket) {
                int actual_port = us_socket_local_port(wSSL? 1: 0, (us_socket_t*)socket);
                assert((!port && actual_port) || (port == actual_port));
                fmt::print(ctx.out(), "Serving over HTTP{} on port {}\n", wSSL? "S": "", actual_port);
            }
        }
    );
```

Which results in:
```
D:\>acme.exe
Serving over HTTP on port 8888
^C
D:\>acme.exe --port 0
Serving over HTTP on port 58196
^C
D:\>acme.exe --port 0
Serving over HTTP on port 58197
```
